pkgs/development/python-modules/dnf-plugins-core/default.nix

```diff
@@ -0,0 +1,118 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+  # dependencies
+, cmake
+, dateutil
+, dbus-python
+, dnf4
+, gettext
+, libcomps
+, libdnf
+, python
+, rpm
+, sphinx
+, systemd
+}:
+
+let
+  pyMajor = lib.versions.major python.version;
+in
+
+buildPythonPackage rec {
+  pname = "dnf-plugins-core";
+  version = "4.4.3";
+  format = "other";
+
+  outputs = [ "out" "man" ];
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = "dnf-plugins-core";
+    rev = version;
+    hash = "sha256-YEw8REvK2X7mBg9HDI6V2p8QtZ3TJh4Dzn8Uuhfbrgo=";
+  };
+
+  patches = [
+    ./fix-python-install-dir.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "@PYTHON_INSTALL_DIR@" "$out/${python.sitePackages}" \
+      --replace "SYSCONFDIR /etc" "SYSCONFDIR $out/etc" \
+      --replace "SYSTEMD_DIR /usr/lib/systemd/system" "SYSTEMD_DIR $out/lib/systemd/system"
+    substituteInPlace doc/CMakeLists.txt \
+      --replace 'SPHINX_BUILD_NAME "sphinx-build-3"' 'SPHINX_BUILD_NAME "${sphinx}/bin/sphinx-build"'
+    substituteInPlace libexec/CMakeLists.txt \
+      --replace "DESTINATION libexec" "DESTINATION $out/bin"
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    sphinx
+  ];
+
+  propagatedBuildInputs = [
+    dateutil
+    dbus-python
+    dnf4.py
+    libcomps
+    libdnf
+    rpm
+    systemd
+  ];
+
+  cmakeFlags = [
+    "-DPYTHON_DESIRED=${pyMajor}"
+    "-DWITHOUT_LOCAL=0"
+  ];
+
+  postBuild = ''
+    make doc-man
+  '';
+
+  pythonImportsCheck = [
+    # This is the python module imported by dnf4 when plugins are loaded.
+    "dnfpluginscore"
+  ];
+
+  # Don's use symbolic links so argv[0] is set to the correct value.
+  postInstall = ''
+    # See https://github.com/rpm-software-management/dnf-plugins-core/blob/aee9cacdeb50768c1e869122cd432924ec533213/dnf-plugins-core.spec#L478
+    ln $out/bin/dnf-utils-${pyMajor} $out/bin/dnf-utils
+
+    # See https://github.com/rpm-software-management/dnf-plugins-core/blob/aee9cacdeb50768c1e869122cd432924ec533213/dnf-plugins-core.spec#L487-L503
+    ln $out/bin/dnf-utils $out/bin/debuginfo-install
+    ln $out/bin/dnf-utils $out/bin/needs-restarting
+    ln $out/bin/dnf-utils $out/bin/find-repos-of-install
+    ln $out/bin/dnf-utils $out/bin/repo-graph
+    ln $out/bin/dnf-utils $out/bin/package-cleanup
+    ln $out/bin/dnf-utils $out/bin/repoclosure
+    ln $out/bin/dnf-utils $out/bin/repodiff
+    ln $out/bin/dnf-utils $out/bin/repomanage
+    ln $out/bin/dnf-utils $out/bin/repoquery
+    ln $out/bin/dnf-utils $out/bin/reposync
+    ln $out/bin/dnf-utils $out/bin/repotrack
+    ln $out/bin/dnf-utils $out/bin/yum-builddep
+    ln $out/bin/dnf-utils $out/bin/yum-config-manager
+    ln $out/bin/dnf-utils $out/bin/yum-debug-dump
+    ln $out/bin/dnf-utils $out/bin/yum-debug-restore
+    ln $out/bin/dnf-utils $out/bin/yum-groups-manager
+    ln $out/bin/dnf-utils $out/bin/yumdownloader
```

Comment:
nit: we could use a construct like the following to make this less repetitive. 

```
for bin in debuginfo-install needs-restarting find-repos-of-install repo-graph; do
  ln $out/bin/dnf-utils $out/bin/$bin
done
```
since we can't just copy the commands anyway, I think this could be a good idea